document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java

```diff
@@ -312,15 +321,13 @@ public long total(Query query) {
       map = parseQuery(query.getFilter());
     }
 
-    final DBObject ref = new BasicDBObject(map);
-    return collection.count(ref);
+    return collection.countDocuments(new BasicDBObject(map));
   }
 
   @Override
   public boolean bulkUpsert(Map<Key, Document> documents) {
     try {
-      BulkWriteOperation bulkWriteOperation = collection.initializeUnorderedBulkOperation();
-
+      List<UpdateOneModel<BasicDBObject>> bulkCollection = new ArrayList<>();
```

Comment:
Checked the codecov for this. There's no coverage going through bulkUpsert?

Reply:
Added now.